pkg/controller/application/application_controller.go

```diff
@@ -459,18 +460,24 @@ func (c *Controller) processApplication(app *shipper.Application) error {
 	}
 
 	apputil.SetHighestObservedGeneration(app, highestObserved)
+
+	app.Status.History = apputil.ReleasesToApplicationHistory(appReleases)
+	if err = c.cleanUpReleasesForApplication(app, appReleases); err != nil {
+		return err
+	}
+
```

Comment:
These 3 added blocks are fairly similar to each other. I wonder if it feels like apputil might be the right place for a new method incorporating this logic?